src/NzbDrone.Core/ImportLists/Trakt/User/TraktUserParser.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections.Generic;
+using System.Linq;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.Parser.Model;
+
+namespace NzbDrone.Core.ImportLists.Trakt.User
+{
+    public class TraktUserParser : TraktParser
+    {
+        private readonly TraktUserSettings _settings;
+        private ImportListResponse _importResponse;
+
+        public TraktUserParser(TraktUserSettings settings)
+        {
+            _settings = settings;
+        }
+
+        public override IList<ImportListItemInfo> ParseResponse(ImportListResponse importResponse)
+        {
+            _importResponse = importResponse;
+
+            var listItems = new List<ImportListItemInfo>();
+
+            if (!PreProcess(_importResponse))
+            {
+                return listItems;
+            }
+
+            var jsonResponse = new List<TraktSeriesResource>();
+
+            if (_settings.TraktListType == (int)TraktUserListType.UserWatchedList)
+            {
+                var jsonWatchedResponse = STJson.Deserialize<List<TraktWatchedResponse>>(_importResponse.Content);
+                switch (_settings.TraktWatchedListType)
+                {
+                    case (int)TraktUserWatchedListType.InProgress:
+                        jsonResponse = jsonWatchedResponse.Where(c => c.Seasons.Where(s => s.Number > 0).Sum(s => s.Episodes.Count) < c.Show.AiredEpisodes).SelectList(c => c.Show);
```

Comment:
Since this is comparing to aired episodes, would an unaired upcoming season not be included and thus series would not be added?

Reply:
As I have seen, it follows the logic you mention. For example, Loki: new season premieres on Oct. 5 and I have seen the entire first season. Since the new season hasn't premiered yet, supposedly, "I've seen the series." On the 5th, when the new chapter comes out, if by chance you didn't have the series on Sonarr, it will be added, because it will become an "in progress" series.

This is how Trakt's "In Progress" section works. Right now, Loki does not appear in the list if I filter by completed: https://trakt.tv/users/danimart1991/progress/watched/time/desc